packages/core/src/sync-store/postgres/migrations.ts

```diff
@@ -455,6 +456,73 @@ const migrations: Record<string, Migration> = {
         .execute();
     },
   },
+  "2024_03_20_0_checkpoint_in_logs_table": {
+    async up(db: Kysely<any>) {
+      await db.schema
+        .alterTable("logs")
+        .addColumn("checkpoint", "numeric(75, 0)")
+        .execute();
+      await db.executeQuery(
```

Comment:
Our business logic assumes that for every row in `logs`, there will be a corresponding row in `blocks` (by hash). However, this is not enforced at the database level. So, what would happen here if there were indeed logs without corresponding blocks? This ties into our chat re factory logs.

Reply:
The checkpoint would be null for those. Since we're updating from a join against blocks, any logs without blocks would not get updated in this migration.